plugins/metadata/plugin.go

```diff
@@ -44,6 +44,54 @@ func init() {
 	timeout.Set(boltOpenTimeout, 0) // set to 0 means to wait indefinitely for bolt.Open
 }
 
+// BoltOptions replicates bolt.Options struct, but excludes function fields and adds tags to allow TOML marshalling.
+type BoltOptions struct {
+	// Sets the DB.NoGrowSync flag before memory mapping the file.
+	NoGrowSync bool `toml:"no_grow_sync"`
+	// Do not sync freelist to disk. This improves the database write performance
+	// under normal operation, but requires a full database re-sync during recovery.
+	NoFreelistSync bool `toml:"no_freelist_sync"`
+	// PreLoadFreelist sets whether to load the free pages when opening
+	// the db file. Note when opening db in write mode, bbolt will always
+	// load the free pages.
+	PreLoadFreelist bool `toml:"preload_freelist"`
+	// NoSync sets the initial value of DB.NoSync. Normally this can just be
+	// set directly on the DB itself when returned from Open(), but this option
+	// is useful in APIs which expose Options but not the underlying DB.
+	NoSync bool `toml:"no_sync"`
+	// Mlock locks database file in memory when set to true.
+	// It prevents potential page faults, however
+	// used memory can't be reclaimed. (UNIX only)
+	Mlock bool `toml:"mlock"`
+	// Sets the DB.MmapFlags flag before memory mapping the file.
+	MmapFlags int `toml:"mmap_flags"`
+	// InitialMmapSize is the initial mmap size of the database
+	// in bytes. Read transactions won't block write transaction
+	// if the InitialMmapSize is large enough to hold database mmap
+	// size. (See DB.Begin for more information)
+	InitialMmapSize int `toml:"init_mmap_size"`
+	// PageSize overrides the default OS page size.
+	PageSize int `toml:"page_size"`
+}
+
+func defaultBoltOptions() BoltOptions {
+	opts := bolt.DefaultOptions
+	return BoltOptions{
+		// Reading bbolt's freelist sometimes fails when the file has a data corruption.
+		// Disabling freelist sync reduces the chance of the breakage.
+		// https://github.com/etcd-io/bbolt/pull/1
+		// https://github.com/etcd-io/bbolt/pull/6
+		NoFreelistSync:  true,
+		NoGrowSync:      opts.NoGrowSync,
```

Comment:
Where are the default values defined?

Reply:
Default values are provided by the `bolt` package itself via `bolt.DefaultOptions` (which are used when `nil` value is passed to `bolt.Open`).

Reply:
I meant if the config file does not provide Bolt options, what values do we fill in here?  Yeah, I understand that bolt.DefaultOptions` will be used if `nil` is provided, but we are changing the implementation here _not_ to provide `nil` any more.

Reply:
TOML config uses bolt's defaults. See [here](https://github.com/containerd/containerd/pull/10745/files#diff-9c09b2ff8e763fbd2736030258f48ef2115b4cf95d00dc1e32871cedde707fc0R78).

Reply:
Got it.  Thanks!